providers/implementations/signature/rsa.c

```diff
@@ -725,7 +725,8 @@ static int rsa_digest_signverify_init(void *vprsactx, const char *mdname,
 {
     PROV_RSA_CTX *prsactx = (PROV_RSA_CTX *)vprsactx;
 
-    prsactx->flag_allow_md = 0;
+    if (prsactx != NULL)
+        prsactx->flag_allow_md = 0;
```

Comment:
The rest of the code seems to assume that `vprsactx != NULL`, like the `prsactx->mdctx = EVP_MD_CTX_new();` line.

Reply:
Except that rsa_signature_init does the NULL check and returns an error.